### PR TITLE
COMPAT: get emptyObject with range header

### LINKED ISSUE #526: Ceph test regression: get empty object with range header
Ceph test not passing anymore:

FAIL: s3tests.functional.test_s3.test_ranged_request_empty_object

### PR DESCRIPTION
**Pull request by nicolas2bert**
FIXES #526

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/legacy/tests.js [hunk 1]
```diff
@@ -376,6 +377,55 @@ describe('aws-node-sdk test suite as registered user', function testSuite() {
...
+                }
+                return s3.deleteBucket({ Bucket: bucketEmptyObj }, err => {
+                    if (err) {
+                        return done(new Error(`error deleting bucket: ${err}`));
+                    }
+                    return done();
+                });
+            });
+        });
+        it('should return InvalidRange if getting empty object with range ' +
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/legacy/tests.js]: should return InvalidRange error if get range on empty object

File: tests/functional/aws-node-sdk/test/legacy/tests.js [hunk 2]
```diff
@@ -376,6 +377,55 @@ describe('aws-node-sdk test suite as registered user', function testSuite() {
...
+        'header',
+        done => {
+            const params = {
+                Bucket: bucketEmptyObj,
+                Key: 'emptyobj',
+                Range: 'bytes=1-2',
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/legacy/tests.js]: what if you actually do a range that is 0 bytes?

File: tests/functional/aws-node-sdk/test/legacy/tests.js [hunk 3]
```diff
@@ -376,6 +377,55 @@ describe('aws-node-sdk test suite as registered user', function testSuite() {
         });
     });
 
+    describe('Get empty object with range header', () => {
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/legacy/tests.js]: Get range on empty object

File: tests/functional/aws-node-sdk/test/legacy/tests.js (hunk 2 above)
Comment (nicolas2bert) [inline tests/functional/aws-node-sdk/test/legacy/tests.js]: I have tried bytes=0-0 on AWS - is it what you meant?

Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/legacy/tests.js]: yes

Comment (nicolas2bert) [inline tests/functional/aws-node-sdk/test/legacy/tests.js]: it returns InvalidRange if object empty :)

**Comment by nicolas2bert** [comment]
@ironman-machine ok

**Comment by LaurenSpiegel** [comment]
end to end passed here: http://ci.ironmann.io/gh/scality/Integration/8749

**Pull request merged**